### ISSUE TITLE
Syntax server no longer exits once JDT server is initialized

### ISSUE THREAD

**Issue by rgrunber** [redhat-developer/vscode-java#1928]
Using vscode-java 0.79.0 or newer, and running `watch -n1 -d 'jps -l | grep -v jps'` in some terminal, I noticed that the syntax server no longer exits once the language server has completed its startup.

##### Environment
Linux (Fedora 33)
VS Code 1.55.2

This seems to have started happening with https://github.com/redhat-developer/vscode-java/pull/1894 ( 9f32875a67352487f5c414bb7fef04c9b00af89d ) , but I don't think we do anything with the exit logic. Is there a chance this is a bug in that library ?

- shutdown() is sent by client, and syntax server responds by saving the workspace
- client confirms it received a response from shutdown
- client then sends an exit() request to server
- the SyntaxLanguageServer.exit() on the server side never gets called by the client

Instead, all I see is the following show up in the errors :

```
rejected promise not handled within 1 second: Error [ERR_STREAM_WRITE_AFTER_END]: write after end
stack trace: Error [ERR_STREAM_WRITE_AFTER_END]: write after end
    at writeAfterEnd (_stream_writable.js:266:14)
    at Socket.write (_stream_writable.js:315:5)
    at /home/rgrunber/git/vscode-java/dist/extension.js:2:882887
    at new Promise (<anonymous>)
    at c.write (/home/rgrunber/git/vscode-java/dist/extension.js:2:882805)
    at _.doWrite (/home/rgrunber/git/vscode-java/dist/extension.js:2:872391)
    at /home/rgrunber/git/vscode-java/dist/extension.js:2:872286
```

Debugging this I see it happens for the following data :
`'Content-Length: 33\r\n\r\n'`
`{"jsonrpc":"2.0","method":"exit"}`

**Comment by rgrunber** [comment]
Looks to be an upstream issue https://github.com/microsoft/vscode-languageserver-node/issues/723 .

**Comment by rgrunber** [comment]
As soon as the upstream issue is resolved (scheduled for 3.16.1?) we can remove this option. This should work around the issue until then.

We're basically calling exit() as part of shutdown() if the client sets this property. I've tested this in lightweight and hybrid modes, and it seems to behave correctly.
